tasks/main.yml

```diff
@@ -1,25 +1,110 @@
 # SPDX-License-Identifier: MIT
 ---
-- name: Set platform/version specific variables
-  include_tasks: tasks/set_vars.yml
-
-# Examples of some tasks:
 - name: Ensure required packages are installed
   package:
-    name: "{{ __template_packages }}"
+    name: '{{ __systemd_packages }}'
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
+- name: Deploy required unit files
+  template:
+    src: '{{ item }}.j2'
```

Comment:
The role must ask users to provide a full path with systemd_unit_files right away, and not add `.j2` to it. And then, in the `dest` cut the extension (`.j2` or other) from `{{ item }}`.

It is possible that users have a systemd unit files not as a template but as a regular file, so it won't have .j2 at the end.

Reply:
Maybe we should just add another task and variable that would deal with plain unit files. So we would have `systemd_unit_files` and `systemd_unit_file_templates`.

Reply:
similar to how I did this in the podman role - https://github.com/linux-system-roles/podman/pull/78/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R147 - you can specify the input as `file_src` which is a file to be used with the `copy` module; `template_src` is used with the `template` module; and `file_content` is used to provide the contents of the file inline e.g.

```yaml
- file_content: |
    [section]
    something=other
    ...
````

content is useful for small units where you can do the templating inline and you don't want to have to manage the local file